core/src/main/java/org/apache/accumulo/core/client/admin/TabletInformation.java

```diff
@@ -18,142 +18,74 @@
-  }
-
-  public int getNumWalLogs() {
-    return this.numWalLogs;
-  }
-
-  public long getNumEntries() {
-    return this.numEntries;
-  }
-
-  public String getNumEntries(final boolean humanReadable) {
-    if (humanReadable) {
-      return NumUtil.bigNumberForQuantity(numEntries);
-    }
-    return String.format("%,d", numEntries);
-  }
-
-  public long getSize() {
-    return this.size;
-  }
-
-  public String getSize(final boolean humanReadable) {
-    if (humanReadable) {
-      return NumUtil.bigNumberForSize(size);
-    }
-    return String.format("%,d", size);
-  }
-
-  public String getStatus() {
-    return this.status;
-  }
-
-  public String getLocation() {
-    return this.location;
-  }
-
-  public String getTabletDir() {
-    return this.dir;
-  }
-
-  public TabletHostingGoal getHostingGoal() {
-    return this.goal;
-  }
-
-  @Override
-  public String toString() {
-    return "TabletInformation{tableName='" + tableName + '\'' + ", numFiles=" + numFiles
-        + ", numWalLogs=" + numWalLogs + ", numEntries=" + numEntries + ", size=" + size
-        + ", status='" + status + '\'' + ", location='" + location + '\'' + ", dir='" + dir + '\''
-        + ", tableId=" + tableId + ", tablet=" + getTablet() + ", goal=" + goal + '}';
-  }
-
-  public static final String header =
-      String.format("%-4s %-15s %-5s %-5s %-9s %-9s %-10s %-30s %-5s %-20s %-20s %-10s", "NUM",
-          "TABLET_DIR", "FILES", "WALS", "ENTRIES", "SIZE", "STATUS", "LOCATION", "ID",
-          "START (Exclusive)", "END", "GOAL");
-
-  @Override
-  public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
-    TabletInformation that = (TabletInformation) o;
-    return numFiles == that.numFiles && numWalLogs == that.numWalLogs
-        && numEntries == that.numEntries && size == that.size
-        && Objects.equals(tableName, that.tableName) && Objects.equals(endRow, that.endRow)
-        && Objects.equals(prevEndRow, that.prevEndRow) && Objects.equals(status, that.status)
-        && Objects.equals(location, that.location) && Objects.equals(dir, that.dir)
-        && Objects.equals(tableId, that.tableId) && goal == that.goal;
-  }
-
-  @Override
-  public int hashCode() {
-    return Objects.hash(tableName, endRow, prevEndRow, numFiles, numWalLogs, numEntries, size,
-        status, location, dir, tableId, goal);
-  }
+import org.apache.accumulo.core.data.TabletId;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata.Location;
+import org.apache.hadoop.io.Text;
+
+public interface TabletInformation {
+
+  /**
+   * @return the TableId of the table containing this tablet.
+   */
+  TableId getTableId();
+
+  /**
+   * @return the TabletId for this tablet.
+   */
+  TabletId getTabletId();
+
+  /**
+   * @return the tablet end row.
+   */
+  Text getEndRow();
+
+  /**
+   * @return the previous end row.
+   */
+  Text getStartRow();
```

Comment:
Since this information could all be obtained from TabletId, could drop the methods in this interface.

```suggestion
  /**
   * @return the TabletId for this tablet.
   */
  TabletId getTabletId();
```